### PR TITLE
centralize (and hide) GDC api host and request headers

### PR DESCRIPTION
**Pull request by siosonel**
## Description
Closes https://github.com/stjude/proteinpaint/issues/1079

Pull sjpp/master: the centralized `getHostHeaders()` is in `sjpp/gdc/dataset/gdc.hg38.js`.
Test with GDC data and examples/URLs for matrix, gene expression, bam slicing, SSM lollipop. 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by karishma-gangwani — DISMISSED** [review]
integration tests pass.

**Review by congyu-lu — APPROVED** [review]

**Review by robinpaul85 — APPROVED** [review]

File: server/src/mds3.init.js [hunk 1]
```diff
@@ -261,6 +261,7 @@ export async function validate_termdb(ds) {
 
 	if (tdb.convertSampleId) {
 		if (tdb.convertSampleId.gdcapi) {
+			// !!!! THIS IS NOT EXPORTED BY mds.gdc !!!
```
Comment (xzhou82) [inline server/src/mds3.init.js:264]: delete

File: server/routes/gdc.mafBuild.ts [hunk 2]
```diff
@@ -27,7 +25,12 @@ export const api = {
 function init({ genomes }) {
 	return async (req: any, res: any): Promise<void> => {
 		try {
-			await buildMaf(req.query as GdcMafBuildRequest, res)
+			// g and ds are not used right now, but could be later
```
Comment (xzhou82) [inline server/routes/gdc.mafBuild.ts:28]: delete comment

File: server/src/bam.js [hunk 3]
```diff
@@ -3399,48 +3412,38 @@ req{}
...
 	// decompress: false prevents got from setting an 'Accept-encoding: gz' request header,
 	// which may not be handled properly by the GDC API in qa-uat
 	// per Phil, should only be used as a temporary workaround
 	// Also:
 	// since the expected response is binary data, should not set Accept: application/json as a request header
 	// also no body is submitted with a GET request, should not set a Content-type request header
-	const headers = { compression: false }
-	if (sessionid) headers['Cookie'] = `sessionid=${sessionid}`
-	else headers['X-Auth-Token'] = token
+	const { host, headers } = ds.getHostHeaders()
```
Comment (xzhou82) [inline server/src/bam.js:3443]: getHostHeaders() lacks parameter to supply token/session

File: server/src/bam.js [hunk 4]
```diff
@@ -3399,48 +3412,38 @@ req{}
...
 	return gdc_bam_filenames
 }
 
-async function streamGdcBam2response(res, regionStr, gdcFileUUID, sessionid, token) {
-	const headers = { compression: false } // see comments in get_gdc_bam()
-	if (sessionid) headers['Cookie'] = `sessionid=${sessionid}`
-	else headers['X-Auth-Token'] = token
-	const url = path.join(apihost, '/slicing/view/', gdcFileUUID + '?region=' + regionStr)
+async function streamGdcBam2response(res, regionStr, gdcFileUUID, ds) {
+	const { host, headers } = ds.getHostHeaders()
```
Comment (xzhou82) [inline server/src/bam.js:3429]: lack argument

File: server/src/bam.gdc.js [hunk 5]
```diff
@@ -320,40 +327,47 @@ async function getBamfileByFileId(id, field) {
...
+
+	/* helper to query api
+		filters: non-case filter, e.g. to test if file id is valid, must not be used in case_filters
+		api:
+		returnSize:
+		filter0: case filters, must not be combined with 1st arg
+	*/
+
+	return async function getQueryApi(filters, api, returnSize, filter0) {
+		const { host, headers } = ds.getHostHeaders()
```
Comment (xzhou82) [inline server/src/bam.gdc.js:342]: lack argument

**Review by xzhou82 — APPROVED** [review]
thank you so much this is very helpful. may have done it piecewise to be easier
i can next clean up token/session stuff from bam.*

**Pull request merged**